src/app/content/hooks/locationChange/resolveContent.ts

```diff
@@ -178,14 +178,17 @@ export const getBookInformation = async(
 
   } else if (UNLIMITED_CONTENT) {
     for (const {id, bookVersion} of allReferences) {
+      // Omit references that don't have bookVersion because new pipeline will not support requests without a version
+      if (!bookVersion) { continue; }
```

Comment:
for now we support both, if we didn't support the old one i'm not sure this check would belong here though, what is it here for?

Reply:
If I understand correctly, the `bookVersion` may be `undefined` if a book is not in the rex config and the AC for this issue is to change links to pop alerts in this case

Reply:
`services.archiveLoader.getBookIdsForPage` is returning `Promise<BookReference[]>` and `BookReference` is typed as `{id: string; bookVersion: string | undefined; }`

is this possible that we will not have `bookVersion` at this stage?

Reply:
`getBookIdsForPage` is not used for web pipeline content. at this point the `bookVersion` key is optional, but it should be ok because `services.archiveLoader.book` accepts `bookVersion` as optional